### ISSUE TITLE
Column calculators do not cover all cases

### ISSUE THREAD

**Issue by burimshala** [fin-hypergrid/core#696]
If you try to create a calculator function that references items in the scope, than it fails to execute.

For instance if you have something like : 

```
import CellRenderer from './MyCellRenderer'

//Schema
export default {
 {
   // other schema properties,
   calculator: (row, columnName) => CellRenderer[columnName]
 }
}
```

You will get an error : CellRenderer is not defined

This problems comes from src/lib/toFunction.js . I don't understand the component internals but it seems like it stringifies the calculator function and tries to parse it later.

**Comment by joneit** [comment]
Hello @burimshala, The parsing algorithm is not expecting arrow functions. `function(row, columnName) { return CellRenderer[columnName] }` should work.

Furthermore, while calculators have no specific execution context at this time (just the global object), they could in a future version, and therefore should never be arrow functions (which do not respect the execution context).

Thanks for the feedback. I will add a more descriptive error message to v3.

By the way, there _is_ an actual error in `toFunction` that has been fixed in the 3.0.0 branch. Unfortunately the fix did not make it into this morning's 2.1.3 release. To avoid this error for now (in version <= 2.1.3), make sure calculator functions do not contain more than one right-curly-brace (`}`) character.

Jonathan

**Comment by joneit** [comment]
@burimshala FYI, I've started a `v2.1.4` branch containing the `toFunction` fix I mentioned above ("By the way...") from the v3 branch.

`2.1.4` also contains a logic fix to calculators that likely doesn't affect your usage, which is that the `column.calculator setter` was not calling toFunction although the `column.properties.calculator` setter was. They are now both equivalent as intended.
